### PR TITLE
Fix: NSNumber conversion to long

### PR DESCRIPTION
**Pull request by buenaflor**
Fixes: #87 where converting an NSNumber to Kotlin Long fails and crashes.

### CODE REVIEW

File: sentry-kotlin-multiplatform/src/commonAppleTest/kotlin/io/sentry/kotlin/multiplatform/SentryExceptionTest.kt [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+import platform.Foundation.NSNumber
+import kotlin.test.Test
+
+class SentryExceptionTest {
+
+    @Test
+    fun `SentryException ThreadId NSNumber long conversion`() {
+        val value = "testValue"
+        val type = "type"
+        val cocoaSentryException = CocoaSentryException(value = value, type = type).apply {
```
Comment (marandaneto) [inline sentry-kotlin-multiplatform/src/commonAppleTest/kotlin/io/sentry/kotlin/multiplatform/SentryExceptionTest.kt:14]: You could extract this to:

```
getCocoaSentryException().apply {
...
}
```

File: sentry-kotlin-multiplatform/src/commonAppleMain/kotlin/io/sentry/kotlin/multiplatform/extensions/SentryExceptionExtensions.kt [hunk 2]
```diff
@@ -7,5 +7,5 @@ internal fun CocoaSentryException.toKmpSentryException() = SentryException(
     type = type,
     value = value,
     module = module,
-    threadId = threadId as Long?
+    threadId = threadId?.longLongValue // longLong represents a 64-bit integer like Kotlin Long
```
[the remark this answers is not in the document]
Comment (buenaflor) [reply sentry-kotlin-multiplatform/src/commonAppleMain/kotlin/io/sentry/kotlin/multiplatform/extensions/SentryExceptionExtensions.kt:10]: I added some Foundation tests regarding this, it looks to be alright. I think the stackoverflow issue has something to do with `NSDecimalNumber` conversions.

[the remark this answers is not in the document]
Comment (marandaneto) [reply sentry-kotlin-multiplatform/src/commonAppleMain/kotlin/io/sentry/kotlin/multiplatform/extensions/SentryExceptionExtensions.kt:10]: Can we add a test just to be sure?

[the remark this answers is not in the document]
Comment (buenaflor) [reply sentry-kotlin-multiplatform/src/commonAppleMain/kotlin/io/sentry/kotlin/multiplatform/extensions/SentryExceptionExtensions.kt:10]: yup, I added them in my newest commit in `FoundationTest.kt`

**Pull request merged**